### ISSUE TITLE
[GSoC] Create a sample for notebook to Kubeflow deployment using TensorFlow 2.0 Keras

### ISSUE THREAD

**Issue by terrytangyuan** [kubeflow/examples#774]
**Description**: Create a notebook that illustrates the core Kubeflow CUJ:
* Develop a model in a notebook using 2.0 Keras
* Launch training/hyperparameter tuning jobs on the cluster 
* Deploy the model using KFServing
* Setup CI/CD for the notebook

**Reference**: [The mnist notebooks](https://github.com/kubeflow/examples/tree/master/mnist) provide a good example of how to construct an E2E example notebook to be run on Kubeflow and also setup CI/CD for the notebook.

**Technical skills required**: TensorFlow & Keras, familiarity with Kubeflow and Jupyter

**Type**: Machine Learning / Data Engineering 

**Difficulty**: Medium 

**Mentors**: Yuan Tang (@terrytangyuan) Ce Gao (@gaocegege)

**Comment by kurianbenoy** [comment]
Hi @terrytangyuan, what should we do before GSoC application deadline? Should we send a draft of our GSoC proposal with notebooks showcasing Kubeflow deployment?

What's your specific expectation of a student required to do this task?

**Comment by terrytangyuan** [comment]
No you don’t have to write notebooks yet. Submitting proposal should be sufficient. [Here](https://docs.google.com/document/d/1dnhvxFLV1odqpqdWdujUNNUhVPSykflLy2nLJCz-Yws/edit#)’s  a template you can use to draft your proposal.

**Comment by yashjakhotiya** [comment]
Hi @terrytangyuan. To help me show competency and assurance that I, as a GSoC student, am able to complete this project, if you think of assigning any small task please let me know here. I'll be doing that along with the proposal.

**Comment by siddharthjain1611** [comment]
What all kubeflow features do you plan for this project?

**Comment by MoAmrYehia** [comment]
Hello @terrytangyuan , I am asking about the project number 4. What is expected for samples to be like? how many samples I need? Should I define them now or it is acceptable to be defined later? Thanks in advance!

**Comment by terrytangyuan** [comment]
@MoAmrYehia Please only ask questions in one place as it's very hard for me to track all the questioins in multiple places. I saw that you've asked on Slack and someone already answered your question. 

@siddharthjain1611 Could you be more specific?

**Comment by siddharthjain1611** [comment]
@terrytangyuan Sorry to bother you sir, but I got my answers on my proposal. Thanks for replying.

**Comment by GG-yuki** [comment]
Hi,sir.Sorry to bother you in your busy schedule.
I have 2 questions to ask you as following:
1.Should I build a model that can work on something special (high accuracy) ,or I can just use MNIST as an example?
2.I can't create services in Jupyter.There's something confusing in the insrtuction of KFServing,and someone has the same problem in that Issues( maybe because I'm a Rookie hhh).Anyway,I would try to fix this problem or ask in slack.
Sincerely,sorry to interrupt you.

**Comment by 11fenil11** [comment]
@terrytangyuan @gaocegege I would love to work on it. As I am working since long with Machine Learning and Jupyter Notebook.

**Comment by terrytangyuan** [comment]
@GG-yuki To answer your questions:
1. The model itself is pretty flexible as the main purpose is to show the use of Kubeflow.
2. Yes, you can post a GitHub issue on KFServing repo or ask on its Slack channel if the instruction is not clear.

@11fenil11 Thanks for your interest. Please submit your proposal as [the GSoC deadline](https://developers.google.com/open-source/gsoc/timeline) is very close.

**Comment by 11fenil11** [comment]
I Have shared draft over GSoC website with all my best knowledge about
project😊.

Please give comments & feedbacks.

Thanks & Regards.

**Comment by jlewi** [comment]
It might be useful to use and highlight the lineage tracking features with MLMD.

See kubeflow/website#1959

**Comment by jlewi** [comment]
@terrytangyuan How's this going?

**Comment by terrytangyuan** [comment]
This can be closed now.

/close
